### ISSUE TITLE
ATX builder tries to build nipost with a stale challenge

### ISSUE THREAD

**Issue by poszu** [spacemeshos/go-spacemesh#3917]
## Description
TLDR; A node fails to build an ATX after smapp has been down for over an hour (more than an epoch) because it tries to use a stale nipost challenge.

```logs
2022-12-26T21:06:01.499-0500	INFO	ad66d.atxBuilder   	ready to build first atx	{"node_id": "ad66d4f5cd9f1e55ed856df70bb7855fcb6c65d1ef04ae6ae9f0b760582be759", "module": "atxBuilder", "current_layer": "3783", "current_epoch": "63", "name": "atxBuilder"}
2022-12-26T21:06:01.501-0500	INFO	ad66d.atxBuilder   	using existing atx challenge	{"node_id": "ad66d4f5cd9f1e55ed856df70bb7855fcb6c65d1ef04ae6ae9f0b760582be759", "module": "atxBuilder", "sessionId": "8c5da117-e062-4413-bbd8-25305b794d1f", "current_epoch": "63", "name": "atxBuilder"}
2022-12-26T21:06:01.501-0500	INFO	ad66d.atxBuilder   	new atx challenge is ready	{"node_id": "ad66d4f5cd9f1e55ed856df70bb7855fcb6c65d1ef04ae6ae9f0b760582be759", "module": "atxBuilder", "sessionId": "8c5da117-e062-4413-bbd8-25305b794d1f", "current_epoch": "63", "publish_epoch": "63", "target_epoch": "64", "name": "atxBuilder"}
2022-12-26T21:06:01.501-0500	INFO	ad66d.atxBuilder   	challenge ready	{"node_id": "ad66d4f5cd9f1e55ed856df70bb7855fcb6c65d1ef04ae6ae9f0b760582be759", "module": "atxBuilder", "name": "atxBuilder"}
2022-12-26T21:06:01.502-0500	INFO	ad66d.atxBuilder   	building NIPost	{"node_id": "ad66d4f5cd9f1e55ed856df70bb7855fcb6c65d1ef04ae6ae9f0b760582be759", "module": "atxBuilder", "poet round start": "2022-12-26 21:10:00.498 -0500 EST", "pub_epoch": "63", "deadline time": "2022-12-26T21:05:30.498-0500", "name": "atxBuilder"}
2022-12-26T21:06:01.516-0500	ERROR	ad66d.atxBuilder   	error attempting to publish atx	{"node_id": "ad66d4f5cd9f1e55ed856df70bb7855fcb6c65d1ef04ae6ae9f0b760582be759", "module": "atxBuilder", "sessionId": "8c5da117-e062-4413-bbd8-25305b794d1f", "layer_id": 3783, "epoch_id": 63, "errmsg": "create ATX: failed to build NIPost: builder: didn't receive any poet proof", "name": "atxBuilder"}
```

It doesn't detect that the challenge is stale and it will not be able to build a valid nipost with it. Especially:
- the current time is `2022-12-26T21:06:01`
- "current_epoch": "63"
- "publish_epoch": "63"
- "target_epoch": "64"
- (next) "poet round start": "2022-12-26 21:10:00" - poet round (that would give proof on time to publish in epoch 63) has already started. (The "poet round start" in the logs is for the next round).
- "deadline time": "2022-12-26T21:05:30.498" - the deadline to wait for a poet proof is in the past and hence it terminates immediately.

## Actual Behavior
`atxBuilder` picks up a stale nipost challenge.

## Expected Behavior
`atxBuilder` should determine that the challenge is stale and create a new one.

## Environment
Please complete the following information:
 - OS: 
 
 - Node Version: 
 
 - Smapp version (if applicable): 
 

## Additional Resources 
[spacemesh-log-55dc4325e8d7f8f9824a47d646599420a2cf4516 (1).zip](https://github.com/spacemeshos/go-spacemesh/files/10318520/spacemesh-log-55dc4325e8d7f8f9824a47d646599420a2cf4516.1.zip)

**Comment by poszu** [comment]
@brusherru I'd appreciate if you updated the steps to reproduce and the environment section.

**Comment by dshulyak** [comment]
I think we shouldn't have this problem anymore, i also made several improvements after debugging there. Please reopen if we catch it again
